lib/workflow/stores/mongo.js

```diff
@@ -283,8 +284,15 @@ function mongoStoreFactory(waterline, Promise, Constants, Errors, assert, _) {
                 instanceId: 1
             }
         };
+        var promise = waterline.graphobjects.findAndModifyMongo(query, {}, graph, options);
 
-        return waterline.graphobjects.findAndModifyMongo(query, {}, graph, options);
+        if (isNew) {
+            promise.then(function () {
+                waterline.graphobjects.publishRecord('created', graph);
```

Comment:
@rolandpoulter I think what we needed here was an 'on.events' message (see https://github.com/RackHD/on-core/blob/master/lib/protocol/events.js#L135) rather than another event on the waterline AMQP exchange?